src/internal/orm/tx.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package orm
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/astaxie/beego/orm"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+var randNumberString = func(n int) string {
+	letterBytes := []byte("1234567890")
+
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letterBytes[rand.Intn(len(letterBytes))]
+	}
+
+	return string(b)
+}
+
+var getTimestamp = func() int64 {
+	return time.Now().Unix()
+}
+
+func getSavepoint() string {
```

Comment:
Seems the algorithm you did is just creating a random serial, why not directly use the UUID?

Reply:
Yes, uuid is uniq and can be used as a savepoint name.

But I want the savepoint name orderly, maybe I should change the last part to auto increment number.

Reply:
Changed to use uuid